### PR TITLE
[CDF-20921, CDF-20920] Fix nested module folder

### PR DESCRIPTION
**Pull request by doctrino**
# Description

Please describe the change you have made.

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/cdf-project-templates/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/cdf-project-templates/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/cdf-project-templates/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/cdf_tk/templates/_templates.py [hunk 1]
```diff
@@ -376,7 +376,13 @@ def process_config_files(
             content = replace_variables(content, local_config)
             filename = create_file_name(filepath, number_by_resource_type)
 
-            destination = build_dir / filepath.parent.name / filename
+            try:
+                _, resource_folder = module_from_path(filepath, return_resource_folder=True)
```
Comment (gregertw) [inline cognite_toolkit/cdf_tk/templates/_templates.py:380]: This seems like a weird overloading of functionality for module_from_path that is not quite intuitive?

**Review by gregertw — APPROVED** [review]
Approving, but I'm a bit unsure how the function code will support the nested modules and you haven't touched that code?

File: cognite_toolkit/cdf_tk/templates/_templates.py (hunk 1 above)
Comment (doctrino) [reply cognite_toolkit/cdf_tk/templates/_templates.py:380]: Good point, making a separate utility function.

**Comment by doctrino** [comment]
> Approving, but I'm a bit unsure how the function code will support the nested modules and you haven't touched that code?

You have hardcoded the resource folder in that module, so it is already covered :)

**Pull request merged**